lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1041,6 +1066,19 @@ def _check_certs_expiration(self, event: UpdateStatusEvent) -> None:
             Scope.UNIT, "certs_exp_checked_at", datetime.now().strftime(date_format)
         )
 
+    def _scrape_config(self) -> List[Dict]:
+        """Generates the scrape config as needed."""
+        pwd = self.secrets.get(Scope.APP, self.secrets.password_key(COSUser))
+        return [
+            {
+                "metrics_path": "/_prometheus/metrics",
+                "static_configs": [{"targets": [f"{self.unit_ip}:{COSPort}"]}],
+                "tls_config": {"insecure_skip_verify": True},
```

Comment:
is it possible to send a CA instead of skip CA check?

Reply:
Hi @delgod, that is used between OpenSearch and cos-agent, both running in the same node. In any case, we will have to work on it in two sides: (1) to check the COS relation, if we receive we can define a certificate via relation; and (2) add that cert to the JVM truststore.

I am putting an Epic to revisit our certificate / passwords story in OpenSearch, I'd advise we consider double-checking COS as part of that Epic, wdyt?
I've created an issue: DPE-3141, to look into that more specifically in the coming weeks.

Reply:
We have changed this to:

```
"tls_config": {"ca": <Opensearch CA CERT>}
```